### PR TITLE
Prompt about losing changes before leaving page if current form is dirty

### PR DESCRIPTION
**Pull request by symetryn**

### CODE REVIEW

File: mintme/src/components/form/CollectibleForm.tsx [hunk 1]
```diff
@@ -91,6 +92,8 @@ export const CollectibleForm = (props: FormProps) => {
     handleRedirect()
   }, [handleRedirect, isSetLoading])
 
+  useRouteChangePrompt(dirty)
```
Comment (saqadri) [inline mintme/src/components/form/CollectibleForm.tsx:95]: Does this need to get reset to false ever?

Comment (symetryn) [reply mintme/src/components/form/CollectibleForm.tsx:95]: Yes when ever we submit/redirect to signin this needs to be false

File: mintme/src/hooks/useRouteChangePrompt.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { useRouter } from "next/router"
+import { useEffect } from "react"
+
+export const useRouteChangePrompt = (isActive) => {
```
[the remark this answers is not in the document]
Comment (symetryn) [reply mintme/src/hooks/useRouteChangePrompt.tsx:4]: yes it does

**Review by an1va — COMMENTED** [review]
Yep, I've had this happen before. Sometimes, the redirect part happens and then the signout event just gets dropped.

**Pull request merged**